extensions/BugModal/web/new_bug.js

```diff
@@ -1,30 +1,78 @@
 var initial = {}
 var comp_desc = {}
+var product_name = '';
+
+var component_load = function(product) {
+    $('#product-throbber').show();
+    $('#component').attr('disabled', true);
+    bugzilla_ajax(
+        {
+            url: 'rest/bug_modal/product_info?product=' + encodeURIComponent(product)
+        },
+        function(data) {
+            $('#product-throbber').hide();
+            $('#component').attr('disabled', false);
+            $('#comp_desc').text('Select a component to read its description.');
+            var selectize = $("#component")[0].selectize;
+            selectize.clear();
+            selectize.clearOptions();
+            selectize.load(function(callback) {
+                callback(data.components)
+            });
+
+            for (var i in data.components)
+                comp_desc[data.components[i]["name"]] = data.components[i]["description"];
+
+            selectize = $("#version")[0].selectize;
+            selectize.clear();
+            selectize.clearOptions();
+            selectize.load(function(callback) {
+                callback(data.versions);
+            });
+        },
+        function() {
+            alert("Network issues. Please refresh the page and try again");
+        }
+    );  
+}
 
 $(document).ready(function() {
+    var current_hash = window.location.hash;
     bugzilla_ajax(
             {
                 url: 'rest/bug_modal/initial_field_values'
             },
             function(data) {
                 initial = data
+                if (current_hash) {
+                    for (product in initial.products) {
+                        if (initial.products[product].name === current_hash.substr(1)) {
+                            product_name = current_hash.substr(1);
+                            var product_node = document.createElement("input");
+                            $("#product_wrap").html('<input name="product" type="hidden" id="product" value="' + product_name + '"><h3 style="padding-left:20px;">'+ product_name +'</h3>');
```

Comment:
This could lead to XSS (maybe self-xss)

much safer:

```javascript
$("#product_wrap").html('<input name="product" type="hidden" id="product"><h3 id="product_name_heading"></h3>')
$("#product").value(product_name);
$("#product_name_heading").text(product_name);
```